lib/hooks/useAddNetworkToMetamask.js

```diff
@@ -0,0 +1,34 @@
+import { WalletContext } from 'lib/components/WalletContextProvider'
+import { NETWORK_DATA } from 'lib/utils/networks'
+import { useContext, useMemo } from 'react'
+
+export const useAddNetworkToMetamask = (chainId) => {
+  const walletContext = useContext(WalletContext)
+
+  // TODO: hide for rinkeby, kovan etc
+  console.log(walletContext)
+
+  return useMemo(() => {
```

Comment:
Do you think this needs useMemo or would benefit of it? I'm wary of using it too much when it's unnecessary unless it actually shows a reasonable performance improvement.

Reply:
https://blog.logrocket.com/rethinking-hooks-memoization/ yeah might be over using it